src/table/stories/index.stories.js

```diff
@@ -101,15 +104,96 @@ storiesOf('Table', module)
       </Card>
     );
   })
-  .add('isLoading', () => (
-    <Card>
-      <CardSegment vertical>
-        <Table
-            components={components}
-            headers={TABLE_HEADERS}
-            isLoading
-            rowsPerPageOptions={[5, 20, 50]}
-            paginated />
-      </CardSegment>
-    </Card>
-  ));
+  .add('isLoading', () => {
+    const [data, setData] = useState([]);
+    const [isLoading, setLoading] = useState(true);
+
+    useEffect(() => {
+      const timeout = setTimeout(() => {
+        setLoading(false);
+        setData(TABLE_DATA);
+      }, 1000);
+
+      return () => {
+        clearTimeout(timeout);
+      };
+    }, [data]);
+
+    return (
+      <Card>
+        <CardSegment vertical>
+          <Table
+              components={components}
+              headers={TABLE_HEADERS}
+              data={data}
+              isLoading={isLoading}
+              rowsPerPageOptions={rowsPerPageOptions}
+              paginated />
+        </CardSegment>
+      </Card>
+    );
+  })
+  .add('Event handlers', () => {
+    const onPageChange = action('onPageChange');
+    const onSort = action('onSort');
+
+    return (
+      <Card>
+        <CardSegment vertical>
+          <Table
+              headers={TABLE_HEADERS}
+              data={TABLE_DATA}
+              rowsPerPageOptions={rowsPerPageOptions}
+              onPageChange={onPageChange}
+              onSort={onSort}
+              paginated />
+        </CardSegment>
+      </Card>
+    );
+  })
+  .add('Exact & totalRows', () => {
+    const [data, setData] = useState(TABLE_DATA.slice(0, 5));
+    const [isLoading, setLoading] = useState(true);
+
+    useEffect(() => {
+      const timeout = setTimeout(() => {
+        setLoading(false);
+      }, 1000);
+
+      return () => {
+        clearTimeout(timeout);
+      };
+    }, []);
+
+    const onPageChange = ({ start, rowsPerPage }) => {
+      setLoading(true);
+      setTimeout(() => {
+        setLoading(false);
+        setData(TABLE_DATA.slice(start, start + rowsPerPage));
+      }, 1000);
+    };
+
+    return (
+      <Card>
+        <CardSegment vertical>
+          <h1>exact + totalRows</h1>
+          <p>
+            {`
+              Use these properties if you only have the exact data to show per page, but still
+              want to suggest the ability to paginate. Use the onPageChange/onSort callbacks to
+              determine when to fetch fresh data.
```

Comment:
Oh I see—so by `exact` data, you mean that on each page data, you're fetching fresh data?

Reply:
That's the intention. You don't always want to fetch everything that could be described in a table at first render. If you're trying to show a table that has every issue, you shouldn't have to actually fetch all those issues - just the ones you're currently looking at. It used to be that the table would paginated based on all the data you have passed in. Now you can make the table show that there are 1000 rows while maintaining the same level of pagination, while only having the necessary data to show one page at a time.

exposing some callbacks with some basic overrides seems less misleading than making an "async" version of the table.